src/indexer/utils/evm/index.ts

```diff
@@ -226,14 +227,35 @@ export async function saveDepositLogs(
   transferRepository: TransferRepository,
   depositRepository: DepositRepository,
   transferMap: Map<string, string>,
+  coinMarketCapService: CoinMarketCapService,
+  sharedConfig: SharedConfig,
 ): Promise<void> {
   let transfer = await transferRepository.findTransfer(decodedLog.depositNonce, Number(decodedLog.fromDomainId), Number(decodedLog.toDomainId))
+  const { amount, fromDomainId } = decodedLog
+
+  const currentDomain = sharedConfig.domains.find(domain => domain.id == parseInt(fromDomainId))
+
+  const currentResource = currentDomain!.resources.find(resource => resource.resourceId == decodedLog.resourceID)
+
+  const tokenSymbol = currentResource?.symbol
+  const resourceType = currentResource?.type
+
+  let amountInUSD: number | null
+
+  try {
+    amountInUSD = await coinMarketCapService.getValueInUSD(amount, tokenSymbol!)
+  } catch (error) {
+    logger.error((error as Error).message)
+    amountInUSD = resourceType !== "fungible" ? null : 0
```

Comment:
If it is not fungible might as well check that first and skip fetching value if it is not.

Reply:
you are correct. Changed